src/mlpack/core/data/load_impl.hpp

```diff
@@ -410,6 +410,77 @@ bool LoadCategorical(const std::string& filename,
   return true;
 }
 
+template<typename MatType>
+bool Load(const std::vector<std::string>& filesname,
+          MatType& matrix,
+          const TextOptions& opts)
+{
+  TextOptions copyOpts(opts);
+  return Load(filesname, matrix, copyOpts);
+}
+
+template<typename MatType>
+bool Load(const std::vector<std::string>& filesname,
+          MatType& matrix,
+          TextOptions& opts)
+{
+  bool success;
+  MatType tmp;
+  arma::field<std::string> firstHeaders;
+  for (size_t i = 0; i < filesname.size(); ++i)
+  {
+    success = Load(filesname.at(i), matrix, opts); 
+    if (opts.HasHeaders())
+    {
+      if (i == 0)
+        firstHeaders = opts.Headers();
+      else
+      {
+        arma::field<std::string> headers = opts.Headers();
+        if (firstHeaders.at(0) != headers.at(0))
```

Comment:
```suggestion
        if (firstHeaders != headers)
```
Why not check all the headers?